File: backend/src/main/java/com/ssafy/cheket/service/queue/QueueService.java
package com.ssafy.cheket.service.queue;

import com.ssafy.cheket.dto.queue.response.QueueEnterResponse;
import com.ssafy.cheket.dto.queue.response.QueueInfoResponse;
import com.ssafy.cheket.dto.queue.response.QueueSeatEnterResponse;

public interface QueueService {

    // 대기열 진입
    QueueEnterResponse enterQueue(Long userId, Long showId, Long sessionId);

    // 대기열 정보 조회
    QueueInfoResponse getQueueInfo(Long userId, Long showId, Long sessionId, String queueToken);

    // 대기열 이탈
    void leaveQueue(Long userId, Long showId, Long sessionId, String queueToken);

    // 좌석 선택 진입
    QueueSeatEnterResponse enterSeatSelection(Long userId, Long showId, Long sessionId, String queueToken);
}

File: frontend/apps/customer-app/android/app/src/main/java/com/ssafy/cheket/core/network/service/QueueService.kt
package com.ssafy.cheket.core.network.service

import com.ssafy.cheket.core.network.dto.*
import retrofit2.http.*

interface QueueService {

    /** 대기열 진입 */
    @POST("api/v1/shows/{showId}/sessions/{sessionId}/queue")
    suspend fun enterQueue(
        @Path("showId") showId: Long,
        @Path("sessionId") sessionId: Long,
    ): ApiResponse<QueueEntryResponse>

    /** 대기열 상태 폴링 (Queue-Token 헤더 필요) */
    @GET("api/v1/shows/{showId}/sessions/{sessionId}/queue/status")
    suspend fun getQueueStatus(
        @Path("showId") showId: Long,
        @Path("sessionId") sessionId: Long,
        @Header("Queue-Token") queueToken: String,
    ): ApiResponse<QueueInfoResponse>

    /** 대기열 이탈 (Queue-Token 헤더 필요) */
    @DELETE("api/v1/shows/{showId}/sessions/{sessionId}/queue")
    suspend fun leaveQueue(
        @Path("showId") showId: Long,
        @Path("sessionId") sessionId: Long,
        @Header("Queue-Token") queueToken: String,
    ): ApiResponse<Unit>

    /** 좌석 선택 진입 (ACTIVE 상태에서 호출, Queue-Token 헤더 필요) */
    @POST("api/v1/shows/{showId}/sessions/{sessionId}/queue/enter")
    suspend fun enterSeatSelection(
        @Path("showId") showId: Long,
        @Path("sessionId") sessionId: Long,
        @Header("Queue-Token") queueToken: String,
    ): ApiResponse<QueueSeatEnterResponse>
}
